src/actions/sources.js

```diff
@@ -247,6 +247,11 @@ export function selectLocation(location: Location, tabIndex: string = "") {
 
     dispatch(addTab(source.toJS(), 0));
 
+    if (source.get("loadedState") == "unloaded") {
+      if (!location.line) location.line = 1;
+      if (!location.column) location.column = 1;
```

Comment:
So lets see here:

1. user wants to go to line 23 / col 47, we shouldn't change the location
2. user doesn't specify a line or column, we default to 1/1

- why is this different for loaded / unloaded sources?

Reply:
A loaded source might already have a state in itself, e.g. the user scrolled the window to line 254 manually and then switched tabs or closed tabs.